### ISSUE TITLE
Help! Cannot fillRect

### ISSUE THREAD

**Issue by siauderman** [melonjs/melonJS#992]
Sorry, I didn't want to disturb you again but I'm facing a very basic problem. I can't fillRect and I can't figure out why from the documentation.

Example below (complete black screen in Chrome, the bounds output in the console are correct):

```
<html>
<head>
	<title>MelonJS Text pointerdown not working</title>
	<script src="https://cdn.jsdelivr.net/npm/melonjs@7.1.1/dist/melonjs.js"></script>
	<script>
		let menu=me.Stage.extend({
			init:function(){
				this._super(me.Stage,'init');
			},
			onResetEvent:function(){
				let entity=me.Entity.extend({
					init:function(){
						let width=me.game.viewport.getWidth();
						let height=me.game.viewport.getHeight();
						this._super(me.Entity,'init',[
							width>height?width*0.8:0,
							width>height?0:height*0.8,
							{
								width:width>height?0.2*width:width,
								height:width>height?height:0.2*height
							}
						]);
						this.floating=true;
					},
					draw:function(renderer){
						renderer.setColor(new me.Color(255,0,0,1));
						let bounds=this.getBounds();
						console.log(bounds.left,bounds.top,bounds.width,bounds.height);
						renderer.fillRect(bounds.left,bounds.top,bounds.width,bounds.height);
						renderer.strokeRect(bounds.left,bounds.top,bounds.width,bounds.height);
					}
				});
				me.game.world.addChild(new entity());
			}
		});
		me.device.onReady(function(){
			me.video.init(document.body.clientWidth,document.body.clientHeight,{
				wrapper:'screen',
				scale:'auto',
				scaleMethod:'flex'
			});
			me.loader.preload([],function(){
				me.state.set(me.state.MENU,new menu());
				me.state.change(me.state.MENU);
			});
		});
	</script>
	<style>
		#screen{
			position:absolute;
			left:0;
			top:0;
			width:100%;
			height:100%;
		}
	</style>
</head>
<body>
<div id="screen"></div>
</div>
</body>
</html>
```

**Comment by obiot** [comment]
the way melonJS works is that it only refresh the display (canvas) If any of the visible object was updated (animation, position change, physic collision, etc...), and this is done on an object basis by either returning true or false from the object update method.

in your case here, you only have one object that "does not do anything", so two to fix it :

1. add an update method (call the parent one) and return true;
2. call me.game.repaint after adding the child to the world container

note that if you are only doing basic drawing, you should rather use me.Renderable as a base object, me.Entity is a much more complex one (basically a me.Renderable with multiple me.Sprite Childs and a me.Body for physic) and kind of an overkill just to draw a rectangle.

Note: this "issue" was addressed in the latest 8.0, as now every time an object is either added or removed from the game word a whole display redraw is done.

**Comment by obiot** [comment]
(closing this one therefore, as not really an issue)
